cstrml/swork/src/lib.rs

```diff
@@ -749,19 +749,123 @@ impl<T: Trait> Module<T> {
         deleted_files: &Vec<(MerkleRoot, u64)>,
         sig: &SworkerSignature
     ) -> bool {
-        api::crypto::verify_work_report_sig(
-            curr_pk,
-            prev_pk,
-            block_number,
-            block_hash,
-            reserved,
-            used,
-            srd_root,
-            files_root,
-            added_files,
-            deleted_files,
-            sig
-        )
+        // 1. Encode
+        let block_number_bytes = Self::encode_u64_to_string_vec_u8(block_number);
+        let reserved_bytes = Self::encode_u64_to_string_vec_u8(reserved);
+        let used_bytes = Self::encode_u64_to_string_vec_u8(used);
+        let added_files_bytes = Self::encode_files(added_files);
+        let deleted_files_bytes = Self::encode_files(deleted_files);
+
+
+        // 2. Construct work report data
+        //{
+        //    curr_pk: SworkerPubKey,
+        //    prev_pk: SworkerPubKey,
+        //    block_number: u64, -> Vec<u8>
+        //    block_hash: Vec<u8>,
+        //    free: u64, -> Vec<u8>
+        //    used: u64, -> Vec<u8>
+        //    free_root: MerkleRoot,
+        //    used_root: MerkleRoot,
+        //    added_files: Vec<(MerkleRoot, u64)>, -> Vec<u8>
+        //    deleted_files: Vec<(MerkleRoot, u64)>, -> Vec<u8>
+        //}
+        let data: Vec<u8> = [
+            &curr_pk[..],
+            &prev_pk[..],
+            &block_number_bytes[..],
+            &block_hash[..],
+            &reserved_bytes[..],
+            &used_bytes[..],
+            &srd_root[..],
+            &files_root[..],
+            &added_files_bytes[..],
+            &deleted_files_bytes[..]
+        ].concat();
+
+        api::crypto::verify_p256_sig(curr_pk, &data, sig)
+        // api::crypto::verify_work_report_sig(
+        //     curr_pk,
+        //     prev_pk,
+        //     block_number,
+        //     block_hash,
+        //     reserved,
+        //     used,
+        //     srd_root,
+        //     files_root,
+        //     added_files,
+        //     deleted_files,
+        //     sig
+        // )
+    }
+
+    // 127 -> "127" -> 49 50 55
+    fn encode_u64_to_string_vec_u8(number: u64) -> Vec<u8> {
+        let mut value = number;
+        let mut encoded_number: Vec<u8> = [].to_vec();
+        loop {
+            encoded_number.push((value%10) as u8 + 48u8); // "0" is 48u8
+            value /= 10;
+            if value == 0 {
+                break;
+            }
+        }
+        encoded_number.reverse();
+        encoded_number
+    }
+
+    fn encode_file_root(fs: Vec<u8>) -> Vec<u8> {
```

Comment:
Put directly use in front of `utils function`?

Reply:
change the order of these util

Reply:
move it into one util file